storage/storage.cpp

```diff
@@ -168,6 +168,11 @@ Country const & Storage::CountryByIndex(TIndex const & index) const
 
 void Storage::GetGroupAndCountry(TIndex const & index, string & group, string & country) const
 {
+  // @TODO(bykoianko) This method can work faster and more correctly.
+  // 1. You need parrents (group) - use m_countries "tree".
+  // 2. You need country - it's TIndex now.
+  // 3. You need to use twine instead of FileName2FullName and FullName2GroupAndMap
```

Comment:
I've rewritten it in English:

  // @TODO(bykoianko) This method can work faster and more correctly.
  // 1. To get id for filling group paramter it's better to use take the parent 
  //    of index parameter in m_countries tree. Just fill gpoup 
  //    with its parent name if valid.
  // 2. Use index as id for filling country paramter.
  // 3. To translate the ids got in (1) and (2) into strings for filling group and country
  //    use platform/get_text_by_id subsystem based on twine.